heroic-core/src/main/java/com/spotify/heroic/HeroicCore.java

```diff
@@ -260,21 +268,59 @@ public HeroicCoreInstance newInstance() throws Exception {
 
     @SuppressWarnings("unchecked")
     private void enableTracing(Map<String, Object> config) throws IOException {
-        final Object stackdriverConfig = config.get("stackdriver");
-        if (stackdriverConfig != null) {
-            final Map<String, String> configMap = (Map) stackdriverConfig;
-            final String projectId = configMap.get("project");
+        final Object lightstepConfig = config.get("lightstep");
+        if (lightstepConfig != null) {
+            final Map<String, String> configMap = (Map) lightstepConfig;
+            final String collectorHost = configMap.getOrDefault("collectorHost",
+                DEFAULT_COLLECTOR_HOST);
+            final String accessToken = configMap.getOrDefault("accessToken", DEFAULT_LIGHTSTEP_TOKEN);
+            final Integer reportingIntervalMs = (Integer) config.getOrDefault("reportingIntervalMs",
+                DEFAULT_LIGHTSTEP_REPORTING_MS);
+            final Integer maxBufferedSpans = (Integer) config.getOrDefault("maxBufferedSpans",
+                DEFAULT_LIGHTSTEP_MAX_SPANS);
+
+
+            if (collectorHost == null) {
+                throw new IllegalArgumentException("Configuration for Lightstep is incomplete");
+            }
+
+            final Options.OptionsBuilder optionsBuilder = new Options.OptionsBuilder()
+                .withAccessToken(accessToken)
+                .withMaxReportingIntervalMillis(reportingIntervalMs)
+                .withMaxBufferedSpans(maxBufferedSpans)
+                .withCollectorHost(collectorHost)
+                .withCollectorProtocol("http")
+                .withCollectorPort(8282)
+                .withComponentName("heroic");
+
+            final Options options;
+            try {
+                options = optionsBuilder.build();
+            } catch (MalformedURLException e) {
+                throw new IllegalArgumentException("Malformed configuration for lightstep.", e);
+            }
+
+            log.info("Setting up Lightstep tracing");
+            LightStepTraceExporter.createAndRegister(new JRETracer(options));
+
+        }
 
-            if (projectId == null) {
-                throw new IllegalArgumentException("Configuration for stackdriver is incomplete");
+        final Object signalfxConfig = config.get("signalfx");
+        if (signalfxConfig != null) {
+            final Map<String, String> configMap = (Map) signalfxConfig;
+            final String region = configMap.get("region");
+            final String defaultEndpoint = configMap.getOrDefault(format("http://signalfxgateway.services"
```

Comment:
this forces anyone using heroic to have a similar naming convention for the signalfx gateway. it would be better to follow a similar pattern to lightstep and just have `collectorHost` or something similar

Reply:
also since this is a required option there should be no default and you can throw an exception